crates/polars-plan/src/dsl/meta.rs

```diff
@@ -18,14 +18,19 @@ impl MetaNameSpace {
             Some(s) => s,
         };
         let mut arena = Arena::with_capacity(8);
-        let expr = to_expr_ir(self.0, &mut arena, schema)?;
-        let ae = arena.get(expr.node());
-        let mut inputs = Vec::with_capacity(2);
-        ae.inputs_rev(&mut inputs);
-        Ok(inputs
-            .iter()
-            .map(|node| node_to_expr(*node, &arena))
-            .collect())
+        let expr_ir = to_expr_ir(self.0, &mut arena, schema)?;
+
+        if expr_ir.has_alias() {
```

Comment:
The problem with this is that it will invent aliases that weren't there before.

Reply:
What do you mean? It should only be set if it had an alias.

Reply:
Hmm.. that was not how it was intended. Let me ping you.